diff --git a/test_regress/driver.pl b/test_regress/driver.pl
index 1e643ec83..7430a9c26 100755
--- a/test_regress/driver.pl
+++ b/test_regress/driver.pl
@@ -1254,8 +1254,8 @@ sub compile {
                                 "TEST_OBJ_DIR=$self->{obj_dir}",
                                 "CPPFLAGS_DRIVER=-D" . uc($self->{name}),
                                 ($self->{verbose} ? "CPPFLAGS_DRIVER2=-DTEST_VERBOSE=1" : ""),
-                                ($param{benchmark} ? "" : "OPT_FAST=-O0"),
-                                ($param{benchmark} ? "" : "OPT_GLOBAL=-O0"),
+                                ($param{benchmark} ? "" : "OPT_FAST=-O1"),
+                                ($param{benchmark} ? "" : "OPT_GLOBAL=-O1"),
                                 "$self->{vm_prefix}",  # bypass default rule, as we don't need archive
                                 ($param{make_flags} || ""),
                         ]);
